sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -176,4 +186,13 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
         typed.avg(_._2), typed.count(_._2), typed.sum(_._2), typed.sumLong(_._2)),
       ("a", 2.0, 2L, 4.0, 4L), ("b", 3.0, 1L, 3.0, 3L))
   }
+
+  test("SPARK-12555 - result should not be corrupted after input columns are reordered") {
+    val ds = sql("SELECT 1279869254 AS a, 'Some String' AS b").as[AggData]
+
+    checkDataset(
+      ds.groupByKey(_.a).agg(NameAgg.toColumn),
```

Comment:
@cloud-fan, so, this issue was not caught in 1.6x codebase (which is existing there), so I don't think adding one test case to avoid regressions in the future will make any harm.